torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -922,28 +922,40 @@ RegisterOperators reg(
          },
          aliasAnalysisSpecialCase()),
      Operator(
-         prim::AutogradAnyNonZero,
+         "prim::AutogradAnyNonZero(...) -> int",
          [](const Node* node) -> Operation {
            size_t num_inputs = node->inputs().size();
-           return [=](Stack& stack) {
+           return [num_inputs](Stack& stack) {
              bool result = false;
-             for (const IValue& t : last(stack, num_inputs)) {
-               if (t.toTensor().defined()) {
-                 result = true;
-                 break;
+             for (const IValue& v : last(stack, num_inputs)) {
+               if (v.isTensor()) {
+                if (v.toTensor().defined()) {
+                  result = true;
+                  break;
+                }
+               } else if (v.isTensorList()) {
+                 for (const at::Tensor& t : v.toTensorListRef()) {
+                   result = true;
+                 }
+                 if (result) {
+                   break;
+                 }
+               } else {
+                 TORCH_INTERNAL_ASSERT(false);
                }
+
              }
              drop(stack, num_inputs);
              stack.emplace_back(result);
              return 0;
            };
-         },
-         aliasAnalysisSpecialCase()),
+         }, aliasAnalysisFromSchema()),
      Operator(
          prim::AutogradAdd,
          [](Stack& stack) {
            at::Tensor a, b;
            pop(stack, a, b);
+           TORCH_INTERNAL_ASSERT(a.defined() || b.defined());
```

Comment:
This assert doesn't have to be true, it is possible both a and b are undefined (the result should be undefined as well).